### PR TITLE
Allow non default installation directories for windows.

### PR DESCRIPTION
**Pull request by w4est**
Changed detection functions to query the installed programs list. This will allow non default installation directories for chrome and firefox on windows.

### CODE REVIEW

File: detect.js [hunk 1]
```diff
@@ -23,12 +24,22 @@ function detectBrowsers(os) {
...
+    if (win32Install.search(FirefoxRegex) || win64Install.search(FirefoxRegex)){
         browsers.firefox = true;
-    } else {
-        console.error("Error: Please install google chrome.");
+    }
+    if (win32Install.search(ChromeRegex) || win64Install.search(ChromeRegex)){
+        browsers.chrome = true;
+    }
+    console.log(browsers);
+    if (!(browsers.firefox || browsers.chrome)) {
```
Comment (rei-kurzweil) [inline detect.js:41]: excellent

File: detect.js [hunk 2]
```diff
@@ -23,12 +24,22 @@ function detectBrowsers(os) {
...
-        browsers.chrome = true;
-    } else if (fs.existsSync("c:/Program Files/Mozilla Firefox")) {
+    const FirefoxRegex = /DisplayName[\r\t\f\v |A-Z|_]+Mozilla Firefox/;
+    const ChromeRegex = /DisplayName[\r\t\f\v |A-Z|_]+Chrome/;
+    // Note: Node by default opens a CMD shell, not a powershell.
+    let win32Install = execSync("reg query HKLM\\Software\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall /s /v DisplayName").toString();
+    let win64Install = execSync("reg query HKLM\\Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall /s /v DisplayName").toString();
+        
+
+    if (win32Install.search(FirefoxRegex) || win64Install.search(FirefoxRegex)){
```
Comment (rei-kurzweil) [inline detect.js:34]: good approach

**Review by rei-kurzweil — APPROVED** [review]
This seems solid to me. Good to merge.

**Comment by rei-kurzweil** [comment]
@w4est is there any way to determine a custom install path and save it, using the registry data?

**Comment by w4est** [comment]
Actually yes. The reg query can display the InstallLocation instead of the DisplayName. Problem is you can't do two queries with /v at once afaik. 

So, have to do a regex on the total output of /s to get it.

**Pull request merged**